### PR TITLE
Updated Ash dependency from 0.24 to 0.26

### LINKED ISSUE #2510: Update Ash dependency from 0.24 to 0.26
We are lagging quite behind. The new versions bring more API exposure, also nicer bitflags.

### PR DESCRIPTION
**Pull request by keringar**
Fixes #2510. A lot of the changes were just renaming various constants that were moved around with no major changes. I went through the diff and checked to make sure everything was updated correctly, but it's possible there could have been a typo somewhere in the 500 or so changes. 

I tested the changes with warden and the examples and nothing appears to be wrong, but I can't be sure since it looks like warden is broken with 7 failing tests on the Vulkan backend. But this appears to be a separate issue from updating Ash since warden is also broken on at least the 0.1 release and maybe earlier.

PR checklist:
- [x] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [x] tested examples with the following backends:
    - vulkan
- [x] `rustfmt` run on changed code

### CODE REVIEW

**Comment by kvark** [comment]
The failures are fixed by #2534 , fwiw

File: src/backend/vulkan/src/conv.rs [hunk 1]
```diff
@@ -208,86 +208,86 @@ pub fn map_mip_filter(filter: image::Filter) -> vk::SamplerMipmapMode {
...
-        Primitive::TriangleList           => vk::PrimitiveTopology::TriangleList,
-        Primitive::TriangleListAdjacency  => vk::PrimitiveTopology::TriangleListWithAdjacency,
-        Primitive::TriangleStrip          => vk::PrimitiveTopology::TriangleStrip,
-        Primitive::TriangleStripAdjacency => vk::PrimitiveTopology::TriangleStripWithAdjacency,
-        Primitive::PatchList(_)           => vk::PrimitiveTopology::PatchList,
+        Primitive::PointList              => vk::PrimitiveTopology::POINT_LIST,
+        Primitive::LineList               => vk::PrimitiveTopology::LINE_LIST,
+        Primitive::LineListAdjacency      => vk::PrimitiveTopology::LINE_LIST_WITH_ADJACENCY,
+        Primitive::LineStrip              => vk::PrimitiveTopology::LINE_STRIP,
+        Primitive::LineStripAdjacency     => vk::PrimitiveTopology::LINE_LIST_WITH_ADJACENCY,
```
Comment (kvark) [inline src/backend/vulkan/src/conv.rs]: this should be `LINE_STRIP_WITH_ADJACENCY`

File: src/backend/vulkan/src/lib.rs [hunk 2]
```diff
@@ -395,21 +401,18 @@ impl hal::PhysicalDevice<Backend> for PhysicalDevice {
                 self.instance
                     .0
                     .create_device(self.handle, &info, None)
-                    .map_err(|err| match err {
-                        ash::DeviceError::LoadError(err) => panic!("{:?}", err),
-                        ash::DeviceError::VkError(err) => Into::<result::Error>::into(err),
-                    })
+                    .map_err(|err| Into::<result::Error>::into(err))
```
Comment (kvark) [inline src/backend/vulkan/src/lib.rs]: nit: could just be `map_err(Into::<result::Error>::into)`

File: src/backend/vulkan/src/conv.rs (hunk 1 above)
Comment (keringar) [reply src/backend/vulkan/src/conv.rs]: Fixed

File: src/backend/vulkan/src/lib.rs (hunk 2 above)
Comment (keringar) [reply src/backend/vulkan/src/lib.rs]: Fixed

**Comment by keringar** [comment]
I squashed and ran rustfmt. The failing Windows builds were also fixed. While I don't have access to an Android device to test, I believe the android stuff in `window.rs` has also been correctly updated.

**Comment by kvark** [comment]
Thanks!
Bors r+

**Pull request merged**